### PR TITLE
Check filesystem for mount point

### LINKED ISSUE #17158: Failed to upload files to the mount point of the ufs cluster configured by parameters
**Alluxio Version:**
2.7.1

**Describe the bug**
Mount a ufs path using option `alluxio.underfs.hdfs.configuration`. The mount was successful, but the file upload failed.

**To Reproduce**

1. Mount a ufs path using option `alluxio.underfs.hdfs.configuration`.
2. There is no configuration file on the worker node.
3. Use `copyFromLocal` upload a file and it will be failed.
4. Add configuration files for worker and re-upload the file, it still fails.

**Expected behavior**
After adding the configuration files to the worker, the file can be re-uploaded without restarting the worker node.

**Are you planning to fix it**
yes. 

**Additional context**
Error log:
```
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.IllegalArgumentException: java.net.UnknownHostException: XX
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2051)
        at com.google.common.cache.LocalCache.get(LocalCache.java:3951)
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3974)
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4958)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem.getFs(HdfsUnderFileSystem.java:831)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem.createDirect(HdfsUnderFileSystem.java:287)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem.create(HdfsUnderFileSystem.java:279)
        at alluxio.underfs.ConsistentUnderFileSystem.createNonexistingFile(ConsistentUnderFileSystem.java:46)
        at alluxio.underfs.UnderFileSystemWithLogging$8.call(UnderFileSystemWithLogging.java:224)
        at alluxio.underfs.UnderFileSystemWithLogging$8.call(UnderFileSystemWithLogging.java:221)
        at alluxio.underfs.UnderFileSystemWithLogging.call(UnderFileSystemWithLogging.java:1213)
        at alluxio.underfs.UnderFileSystemWithLogging.createNonexistingFile(UnderFileSystemWithLogging.java:221)
        at alluxio.worker.grpc.UfsFileWriteHandler.createUfsFile(UfsFileWriteHandler.java:166)
        at alluxio.worker.grpc.UfsFileWriteHandler.writeBuf(UfsFileWriteHandler.java:139)
        at alluxio.worker.grpc.UfsFileWriteHandler.writeBuf(UfsFileWriteHandler.java:49)
        at alluxio.worker.grpc.AbstractWriteHandler.writeData(AbstractWriteHandler.java:281)
        at alluxio.worker.grpc.AbstractWriteHandler.lambda$writeDataMessage$1(AbstractWriteHandler.java:179)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at alluxio.worker.grpc.GrpcExecutors$ImpersonateThreadPoolExecutor.lambda$execute$0(GrpcExecutors.java:159)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: java.net.UnknownHostException: XX
        at org.apache.hadoop.security.SecurityUtil.buildTokenService(SecurityUtil.java:444)
        at org.apache.hadoop.hdfs.NameNodeProxiesClient.createProxyWithClientProtocol(NameNodeProxiesClient.java:142)
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:375)
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:309)
        at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:168)
        at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3269)
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:472)
        at org.apache.hadoop.fs.Path.getFileSystem(Path.java:356)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem$1.load(HdfsUnderFileSystem.java:186)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem$1.load(HdfsUnderFileSystem.java:172)
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2278)
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155)
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045)
        ... 21 more
Caused by: java.net.UnknownHostException: XX
        ... 35 more
```

### PR DESCRIPTION
**Pull request by Haoning-Sun**
### What changes are proposed in this pull request?

Fix #17158.

### CODE REVIEW

**Comment by ChunxuTang** [comment]
Feel free to run `mvn checkstyle:checkstyle` to locate the style errors.

File: core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java [hunk 1]
```diff
@@ -168,11 +168,14 @@ private UnderFileSystem getOrAddWithRecorder(AlluxioURI ufsUri,
...
         connectUfs(fs);
-      } catch (IOException e) {
+        tryUseFileSystem(fs, ufsUri.getPath());
+      } catch (Exception e) {
+        mUnderFileSystemMap.remove(key);
         String message = String.format(
             "Failed to perform initial connect to UFS %s: %s", ufsUri, e);
         recorder.record(message);
         LOG.warn(message);
+        throw new RuntimeException(e);
```
Comment (ChunxuTang) [inline core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java:178]: Would you mind explaining why you want to throw the exception after catching it?

File: core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java [hunk 2]
```diff
@@ -185,6 +188,17 @@ private UnderFileSystem getOrAddWithRecorder(AlluxioURI ufsUri,
...
   protected abstract void connectUfs(UnderFileSystem fs) throws IOException;
 
+  /**
+   * To check whether the filesystem is available by calling exists.
+   *
+   * @param fs the filesystem
+   * @param uriPath the UFS path
+   * @throws Exception
+   */
+  private void tryUseFileSystem(UnderFileSystem fs, String uriPath) throws Exception {
```
Comment (ChunxuTang) [inline core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java:198]: As the `uriPath` represents the UFS path, how about renaming it to `ufsUriPath` or `ufsPath`?

**Review by ChunxuTang — COMMENTED** [review]
@Haoning-Sun Thanks for your contribution! Just left some minor suggestions.

File: core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java (hunk 1 above)
Comment (Haoning-Sun) [reply core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java:178]: If don't throw the exception, the created ufs client will be saved by `mMountIdToUfsInfoMap`, but this client may not be available. It affects the operations of the mount point.

Comment (ChunxuTang) [reply core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java:178]: Thanks for the clarification.

File: core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java [hunk 3]
```diff
@@ -168,11 +168,14 @@ private UnderFileSystem getOrAddWithRecorder(AlluxioURI ufsUri,
       mCloser.register(fs);
       try {
         connectUfs(fs);
-      } catch (IOException e) {
+        tryUseFileSystem(fs, ufsUri.getPath());
+      } catch (Exception e) {
+        mUnderFileSystemMap.remove(key);
```
Comment (jja725) [inline core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java:173]: prefer to put ufs into mUnderFileSystemMap only after we verify that ufs is working.

**Review by jja725 — COMMENTED** [review]
Mostly LGTM, leave a minor comment

File: core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java (hunk 3 above)
Comment (Haoning-Sun) [reply core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java]: done.

File: core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java [hunk 4]
```diff
@@ -160,20 +160,21 @@ private UnderFileSystem getOrAddWithRecorder(AlluxioURI ufsUri,
...
             "Failed to perform initial connect to UFS %s: %s", ufsUri, e);
         recorder.record(message);
         LOG.warn(message);
+        throw new RuntimeException(e);
+      }
+      if (mUnderFileSystemMap.putIfAbsent(key, fs) != null) {
+        // This shouldn't occur unless our synchronization is incorrect
+        LOG.warn("UFS already existed in UFS manager");
       }
+      mCloser.register(fs);
```
[the remark this answers is not in the document]
Comment (Haoning-Sun) [reply core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java:177]: I got it.

**Review by jja725 — APPROVED** [review]
LGTM, although we change the behavior a little.

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**